python/cudf/cudf/io/parquet.py

```diff
@@ -776,7 +794,12 @@ def __init__(
                     <filename>.parquet
 
         """
-        self.path = path
+        if isinstance(path, str) and path.startswith("s3://"):
+            self.fs_meta = {"is_s3": True, "actual_path": path}
+            self.path = tempfile.TemporaryDirectory().name
+        else:
+            self.fs_meta = {}
+            self.path = path
```

Comment:
Is there a reason that this PR is only adding this feature for `s3`? Can't the same code path be used for `gcs` as well?

Reply:
The reason of adding `s3` only is to unblock kratos team: https://github.com/rapidsai/cudf/issues/10522

Yes, we can add `gcs` too but will need the same amount of testing I have done for s3. Are you okay if I do it in a follow-up PR?

Reply:
The reason for adding `s3` only is to unblock this user: https://github.com/rapidsai/cudf/issues/10522

Yes, we can add `gcs` too but will need the same amount of testing I have done for `s3`. Are you okay if I do it in a follow-up PR?

Reply:
>Yes, we can add gcs too but will need the same amount of testing I have done for s3. Are you okay if I do it in a follow-up PR?

Yes sorry - I should have clarified that GCS support is not a blocker for this PR. I was just curious if there was a larger reason. Makes sense.

Reply:
Opened an issue to track it separately: https://github.com/rapidsai/cudf/issues/10787